HoloLensCommander/HoloLensCommander/Device/DeviceMonitorCommands.cs

```diff
@@ -247,7 +254,27 @@ public Uri GetMixedRealityViewUri()
         /// <returns>Task object used for tracking method completion.</returns>
         public async Task RebootAsync()
         {
-            await this.devicePortal.RebootAsync();    
+            await this.devicePortal.RebootAsync();
+
+            // Force re-establishing the connection.
+            // This is needed in case the device's name has been changed.
+            this.firstContact = false;
+        }
+
+        /// <summary>
+        /// Set the name of the device.
+        /// </summary>
+        /// <param name="name">The new name for the device.</param>
+        /// <returns>Task object used for tracking method completion.</returns>
+        /// <remarks>The name change does not go into effect until the device has been rebooted.</remarks>
+        public async Task<bool> SetDeviceNameAsync(string name)
+        {
+            if (this.MachineName == name) { return false; }
```

Comment:
will replace with string.equals(a,b). my above comment was in reference to string.equals(b).